### PR TITLE
[INFRA] cmake: use seqan3 googletest

### PR DESCRIPTION
**Pull request by marehr**
A better attempt at fixing https://github.com/seqan/app-template/issues/37

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -14,12 +14,10 @@ list (APPEND SEQAN3_EXTERNAL_PROJECT_CMAKE_ARGS "-DCMAKE_BUILD_TYPE=${CMAKE_BUIL
 list (APPEND SEQAN3_EXTERNAL_PROJECT_CMAKE_ARGS "-DCMAKE_INSTALL_PREFIX=${PROJECT_BINARY_DIR}")
 list (APPEND SEQAN3_EXTERNAL_PROJECT_CMAKE_ARGS "-DCMAKE_VERBOSE_MAKEFILE=${CMAKE_VERBOSE_MAKEFILE}")
 set (SEQAN3_TEST_CLONE_DIR "${PROJECT_BINARY_DIR}/vendor/googletest")
-file (MAKE_DIRECTORY ${SEQAN3_TEST_CLONE_DIR}/googletest/include/)
```
Comment (eseiler) [inline test/CMakeLists.txt]: I think we may not need this, because I don't use `INTERFACE_INCLUDE_DIRECTORIES `

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -45,10 +43,11 @@ macro (add_app_test test_filename test_alternative)
 
     # Create the test target.
     add_executable (${target} ${test_filename})
-    target_link_libraries (${target} "${PROJECT_NAME}_lib" seqan3::seqan3 gtest_main)
+    target_link_libraries (${target} "${PROJECT_NAME}_lib" seqan3::seqan3 gtest gtest_main)
```
Comment (eseiler) [inline test/CMakeLists.txt:46]: there are two libs

**Review by eseiler — COMMENTED** [review]
I tried to get it running, I removed gmock for now, but we could just set `BUILD_GMOCK` to `ON`.
It may also work to add the include directory to the interface, but it didn't work for me

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -42,10 +43,11 @@ macro (add_app_test test_filename test_alternative)
 
     # Create the test target.
     add_executable (${target} ${test_filename})
-    target_link_libraries (${target} "${PROJECT_NAME}_lib" seqan3::seqan3 gtest_all pthread)
+    target_link_libraries (${target} "${PROJECT_NAME}_lib" seqan3::seqan3 gtest gtest_main)
 
     # Make seqan3::test available for both cli and api tests.
     target_include_directories(${target} PUBLIC "${SEQAN3_CLONE_DIR}/test/include")
+    target_include_directories(${target} PUBLIC "${SEQAN3_TEST_CLONE_DIR}/googletest/include/")
```
Comment (eseiler) [inline test/CMakeLists.txt:50]: See https://github.com/seqan/seqan3/blob/94a4e2251cc4d813032c7bd6d090970a93f916e8/test/seqan3-test.cmake#L73

though it's a bit different there.

**Comment by eseiler** [comment]
[app-template infra](https://trello.com/c/ERegbv7w/41-app-template-infra)

**Review by MitraDarja — APPROVED** [review]
LGTM. :)

**Review by eseiler — APPROVED** [review]

**Pull request merged**